pytorch_lightning/callbacks/gpu_stats_monitor.py

```diff
@@ -0,0 +1,164 @@
+    is a callback and in order to use it you need to assign a logger in the ``Trainer``.
+
+    Args:
+        memory_utilization: Set to ``True`` to log used, free and percentage of memory
+            utilization at the start and end of each step. Default: ``True``.
+        gpu_utilization: Set to ``True`` to log percentage of GPU utilization
+            at the start and end of each step. Default: ``True``.
+        intra_step_time: Set to ``True`` to log the time of each step. Default: ``False``.
+        inter_step_time: Set to ``True`` to log the time between the end of one step
+            and the start of the next step. Default: ``False``.
+        fan_speed: Set to ``True`` to log percentage of fan speed. Default: ``False``.
+        temperature: Set to ``True`` to log the memory and gpu temperature in degree Celsius.
+            Default: ``False``.
+
+    Example::
+
+        >>> from pytorch_lightning import Trainer
+        >>> from pytorch_lightning.callbacks import GPUStatsMonitor
+        >>> gpu_stats = GPUStatsMonitor() # doctest: +SKIP
+        >>> trainer = Trainer(callbacks=[gpu_stats]) # doctest: +SKIP
+
+    GPU stats are mainly based on `nvidia-smi --query-gpu` command. The description of the queries is as follows:
+
+    - **fan.speed** – The fan speed value is the percent of maximum speed that the device's fan is currently
+      intended to run at. It ranges from 0 to 100 %. Note: The reported speed is the intended fan speed.
+      If the fan is physically blocked and unable to spin, this output will not match the actual fan speed.
+      Many parts do not report fan speeds because they rely on cooling via fans in the surrounding enclosure.
+    - **memory.used** – Total memory allocated by active contexts.
+    - **memory.free** – Total free memory.
+    - **utilization.gpu** – Percent of time over the past sample period during which one or more kernels was
+      executing on the GPU. The sample period may be between 1 second and 1/6 second depending on the product.
+    - **utilization.memory** – Percent of time over the past sample period during which global (device) memory was
+      being read or written. The sample period may be between 1 second and 1/6 second depending on the product.
+    - **temperature.gpu** – Core GPU temperature, in degrees C.
+    - **temperature.memory** – HBM memory temperature, in degrees C.
+
+    """
+
+    def __init__(
+        self,
+        memory_utilization: bool = True,
+        gpu_utilization: bool = True,
+        intra_step_time: bool = False,
+        inter_step_time: bool = False,
+        fan_speed: bool = False,
+        temperature: bool = False
+    ):
+        super().__init__()
+
+        if shutil.which("nvidia-smi") is None:
+            raise MisconfigurationException(
+                'Cannot use GPUStatsMonitor callback because NVIDIA driver is not installed.'
+            )
```

Comment:
can we move this check also to on_train_start, because the code should stay device agnostic so that when we switch from `Trainer(gpus=2`) to `Trainer(gpus=0)` it should just do nothing. This is important because we may want to debug the code locally on a cpu and then push it to a gpu machine for actual training.

Reply:
but perhaps this is too much for this PR. Should we do this in a new one?

Reply:
even if we put this in `on_train_start` and someone wants to debug the code on CPU, even then this Exception will be called whenever .fit will be called.

Reply:
not sure if it is needed at all if you do not have NVIDIA drivers install no GPU is detected and it will fail even sooner that you ask GPU training but you dot have any...

Reply:
then one should remove this callback right away 😁.